### PR TITLE
commitment: Use hash commitments instead of Pedersen

### PR DESCRIPTION
**Pull request by joeykraut**
### Purpose
This PR changes the commitment scheme underlying `Scalar` types to use a hash commitment rather than Pedersen. This is more efficient to compute and allows for simpler batch commit semantics. 

### Testing
- Unit and integration tests pass

### CODE REVIEW

File: src/commitment.rs [hunk 1]
```diff
@@ -93,25 +44,35 @@ impl<C: CurveGroup> HashCommitment<C> {
...
+
         let mut rng = thread_rng();
         let blinder = Scalar::random(&mut rng);
-        let comm = value.fabric.new_gate_op(vec![value.id], move |mut args| {
-            let value: CurvePoint<C> = args.remove(0).into();
+        let ids = values.iter().map(|v| v.id()).collect_vec();
+
+        let comm = fabric.new_gate_op(ids, move |args| {
+            let values = args.into_iter().map(Into::<T>::into);
+            let mut bytes = values.flat_map(|v| v.to_bytes()).collect_vec();
```
Comment (akirillo) [inline src/commitment.rs:73]: nit: could just do `flat_map(ToBytes::to_bytes)` here as well

File: src/commitment.rs [hunk 2]
```diff
@@ -124,6 +85,136 @@ impl<C: CurveGroup> HashCommitmentResult<C> {
...
+
+    /// Verify a commitment to a value
+    async fn verify_comm<T: From<ResultValue<TestCurve>> + Unpin + ToBytes>(
+        comm: HashCommitmentResult<TestCurve, T>,
+    ) -> bool {
+        let comm_res = HashCommitment {
+            blinder: comm.blinder,
+            values: future::join_all(comm.values).await,
+            commitment: comm.commitment.await,
+        };
```
Comment (akirillo) [inline src/commitment.rs:113]: is it worth it to `impl Future for HashCommitmentResult` for this to be more ergonomic (if we do this outside of testing code)?

File: src/algebra/scalar/authenticated_scalar.rs [hunk 3]
```diff
@@ -364,49 +364,47 @@ impl<C: CurveGroup> AuthenticatedScalarResult<C> {
...
-                        peer_mac_share,
-                        peer_commitment,
-                        peer_blinder,
-                    );
-                    mac_checks.push(ResultValue::Scalar(Scalar::from(mac_check)));
+                for (my_mac_share, peer_mac_share) in
+                    izip!(my_comms.into_iter(), peer_mac_checks.into_iter(),)
+                {
+                    let sum_to_zero = my_mac_share + peer_mac_share == Scalar::zero();
+                    mac_checks.push(ResultValue::Scalar(sum_to_zero.into()))
```
Comment (akirillo) [inline src/algebra/scalar/authenticated_scalar.rs:407]: thoughts on re-packaging this into a `verify_mac_check_batch` method (or similar) just so it reads similarly to `open_authenticated_batch` for `CurvePoint`? (just a nit, really)

**Review by akirillo — APPROVED** [review]

File: src/commitment.rs (hunk 2 above)
Comment (joeykraut) [reply src/commitment.rs:113]: We don't do this outside testing code, they're never awaited locally, just shipped off to the recipient

File: src/algebra/scalar/authenticated_scalar.rs (hunk 3 above)
Comment (joeykraut) [reply src/algebra/scalar/authenticated_scalar.rs:407]: I like it

**Pull request merged**